src/components/LayoutTree.tsx

```diff
@@ -364,6 +386,167 @@ const linearizeTree = (
   return thoughts
 }
 
+/** Renders a thought component for mapped treeThoughtsPositioned. */
+const RenderMappedThought = (props: TreeThoughtPositionedComponent) => {
+  const {
+    belowCursor,
+    cliff,
+    depth,
+    env,
+    height,
+    indexChild,
+    indexDescendant,
+    isCursor,
+    isTableCol1,
+    isTableCol2,
+    thoughtKey,
+    leaf,
+    path,
+    prevChild,
+    showContexts,
+    isLastVisible,
+    simplePath,
+    singleLineHeightWithCliff,
+    style,
+    thoughtId,
+    width,
+    autofocus,
+    x,
+    y: _y,
+    index,
+    viewportBottom,
+    treeThoughtsPositioned,
+    bulletWidth,
+    cursorUncleId,
+    setSize,
+    cliffPaddingStyle,
+    dragInProgress,
+    autofocusDepth,
+  } = props
+
+  const [y, setY] = useState(_y)
+
+  useEffect(() => {
+    if (y !== _y) {
+      // When y changes React re-renders the component with the new value of y. It will result in a visual change in the DOM.
+      // Because this is a state-driven change, React applies the updated value to the DOM, which causes the browser to recognize that
+      // a CSS property has changed, thereby triggering the CSS transition.
+      setY(_y)
+    }
+  }, [y, _y])
```

Comment:
Slightly concerning, but won't we trigger an extra render and call to the hook with this, since `y` is a dependency and also gets updated within the effect? Perhaps that extra render is a part of this fix, but I admittedly do not understand.

Reply:
The root cause as we discussed was due to a batched update in React lifecycle, that is why it only animated the first sequence of the renders (top to bottom transition in this case).

useEffect serves a purpose of getting a better control at how we orchestrate its render cycle.